### PR TITLE
[DHF-9] Health Check Implementation for Spruce controller

### PR DESCRIPTION
**Pull request by skt123**
@jackchi 

Reverts [SmartThingsCommunity/SmartThingsPublic#1843](https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/1843)

Please review and merge.

### CODE REVIEW

**Comment by jackchi** [comment]
👍

**Comment by workingmonk** [comment]
@tpmanley does the config change look right to you? it looks like they had max report time less than min report interval. will changing it to default have issues?

**Comment by tpmanley** [comment]
There's nothing in zigbee/ZCL from changing the max report interval like that. With the current config it should only be reporting on/off when the value changes but no more than once a second. With the new value it should also report every 10 minutes even if it hasn't changed.

**Comment by workingmonk** [comment]
thanks @tpmanley 
@jackchi are you ready to merge this? is the QA complete for this change?

**Comment by jackchi** [comment]
Tanuscree is testing this now

File: devicetypes/plaidsystems/spruce-controller.src/spruce-controller.groovy [hunk 1]
```diff
@@ -109,7 +110,7 @@ metadata {
 		//ST release
 		//fingerprint endpointId: '1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18', profileId: '0104', deviceId: '0002', deviceVersion: '00', inClusters: '0000,0003,0004,0005,0006,000F', outClusters: '0003, 0019', manufacturer: 'PLAID SYSTEMS', model: 'PS-SPRZ16-01'
 		//new release
-        fingerprint endpointId: "1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18", profileId: "0104", deviceId: "0002", deviceVersion: "00", inClusters: "0000,0003,0004,0005,0006,0009,000A,000F", outClusters: "0003, 0019", manufacturer: "PLAID SYSTEMS", model: "PS-SPRZ16-01"
+        fingerprint endpointId: "1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18", profileId: "0104", deviceId: "0002", deviceVersion: "00", inClusters: "0000,0003,0004,0005,0006,0009,000A,000F", outClusters: "0003, 0019", manufacturer: "PLAID SYSTEMS", model: "PS-SPRZ16-01", deviceJoinName: "Spruce Irrigation Controller"
```
Comment (jackchi) [inline devicetypes/plaidsystems/spruce-controller.src/spruce-controller.groovy]: It pairs with the older fingerprints not the new one. 
Tanuscree could not pair Spruce controller with this fingerprint @skt123

**Pull request closed without merging**